src/config.ts

```diff
@@ -161,6 +161,10 @@ export class KubeConfig {
                 username: user.username,
             };
         }
+
+        if (cluster && cluster.tlsServerName) {
+            opts.agentOptions = { servername: cluster.tlsServerName } as https.RequestOptions;
```

Comment:
I think this should just be `opts.servername = cluster.tlsServerName` instead of `opts.agentOptions`

Reply:
Ah, looks like you're right. Apologies for the mis-direction.